src/Bridge/Doctrine/Orm/Metadata/Property/DoctrineOrmPropertyNameCollectionFactory.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Bridge\Doctrine\Orm\Metadata\Property;
+
+use ApiPlatform\Core\Exception\ResourceClassNotFoundException;
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyNameCollectionFactoryInterface;
+use ApiPlatform\Core\Metadata\Property\PropertyNameCollection;
+use Doctrine\Common\Persistence\ManagerRegistry;
+
+/**
+ * Fetch inherited properties in case of a SINGLE_TABLE and JOINED doctrine inheritance types.
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+final class DoctrineOrmPropertyNameCollectionFactory implements PropertyNameCollectionFactoryInterface
```

Comment:
I don't really understand the class name, why not something related to what it does ?